sirepo/template/cortex_sql_db.py

```diff
@@ -0,0 +1,86 @@
+"""Cortex db
+
+:copyright: Copyright (c) 2025 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+
+from pykern.pkcollections import PKDict
+from pykern.pkdebug import pkdc, pkdlog, pkdp
+import contextlib
+import pykern.sql_db
+import sqlalchemy
+import sirepo.sim_data
+import pykern.pkio
+
+_BASE = "cortex.sqlite3"
+
+
+def insert_material(parsed, qcall=None):
+    def _values(cols, vals):
+        return PKDict((c, vals[c]) for c in cols if c in vals)
+
+    with _session(qcall=qcall) as s:
+        i = s.insert(
+            "material",
```

Comment:
The insert may fail with an IntegrityError ```UNIQUE constraint failed: material.material_name``` if the same xlsx file is uploaded twice. It should return a better error message to the client in this case.